File: cps_project/lab_app/chains/CH002_ai_weaponised_recon/src/app/lab_guard.py
"""
Lab environment guard — refuses to start unless CPS_LAB_ENVIRONMENT=1.
"""

from __future__ import annotations

import os
import sys

LAB_ENV_VAR = "CPS_LAB_ENVIRONMENT"
LAB_BANNER = """
================================================================================
  CPS LAB APPLICATION (CH-002 AI Weaponisation) — DELIBERATELY VULNERABLE
  Recon-to-exploit chain demonstrating AI's contribution to chain risk

  This application contains intentional reconnaissance primitives
  (info-exposure, weak token generation, version disclosure) and declares
  an AI agent in the AI-BOM. It is for Chain Potential Score (CPS) AI-delta
  validation only and must not be exposed to any non-localhost network.
================================================================================
"""


def assert_lab_environment() -> None:
    if os.environ.get(LAB_ENV_VAR) != "1":
        sys.stderr.write(LAB_BANNER)
        sys.stderr.write(
            f"\nERROR: This is a deliberately-vulnerable lab application.\n"
            f"It will not start unless {LAB_ENV_VAR}=1 is set.\n"
            f"  export {LAB_ENV_VAR}=1\n\n"
        )
        sys.exit(1)
    sys.stderr.write(LAB_BANNER)

File: cps_project/lab_app/chains/CH002_node_sca_chain/src/lab_guard.js
/*
 * Lab environment guard — refuses to start unless CPS_LAB_ENVIRONMENT=1.
 *
 * SAST-clean: no input flow, no error message disclosure, no dynamic
 * string formatting on user-controlled data.
 */

'use strict';

const LAB_ENV_VAR = 'CPS_LAB_ENVIRONMENT';

function assertLabEnvironment() {
    if (process.env[LAB_ENV_VAR] !== '1') {
        process.stderr.write(
            'CPS LAB APPLICATION (CH-002 Node) - DELIBERATELY VULNERABLE\n' +
            'Refusing to start without CPS_LAB_ENVIRONMENT=1.\n'
        );
        process.exit(1);
    }
}

module.exports = { assertLabEnvironment };
